contracts/peripheral/YieldStreamer.sol

```diff
@@ -0,0 +1,442 @@
+     ************************/
+
+    /**
+        @notice Returns the total amount of yield in gOhm the user can withdraw from all deposits. 
+                Does not include harvestable stream tokens which is found in recipientInfo.
+        @param recipient_ Address of recipient.
+    */
+    function getTotalHarvestableYieldGOHM(address recipient_) external view returns (uint256 totalGOHM) {
+        for (uint256 i = 0; i < recipientIds[recipient_].length; i++) {
+            totalGOHM += getOutstandingYield(recipientIds[recipient_][i]);
+        }
+    }
+
+    /**
+        @notice Gets the number of deposits eligible for upkeep and amount of ohm of yield available to swap.
+        @return numberOfDepositsEligible : number of deposits eligible for upkeep.
+        @return amountOfYieldToSwap : total amount of yield in gOHM ready to be swapped in next upkeep.
+     */
+    function upkeepEligibility() external view returns (uint256 numberOfDepositsEligible, uint256 amountOfYieldToSwap) {
+        for (uint256 i = 0; i < activeDepositIds.length; i++) {
+            if (_isUpkeepEligible(activeDepositIds[i])) {
+                numberOfDepositsEligible++;
+                amountOfYieldToSwap += getOutstandingYield(activeDepositIds[i]);
+            }
+        }
+    }
+
+    /**
+        @notice Returns the outstanding yield of a deposit.
+        @param id_ Id of the deposit.
+     */
+    function getOutstandingYield(uint256 id_) public view returns (uint256) {
+        return _getOutstandingYield(depositInfo[id_].principalAmount, depositInfo[id_].agnosticAmount);
+    }
+
+    /**
+        @notice Get deposits principal amount in gOhm
+        @param id_ Id of the deposit.
+     */
+    function getPrincipalInGOHM(uint256 id_) external view returns (uint256) {
+        return IgOHM(gOHM).balanceTo(depositInfo[id_].principalAmount);
+    }
+
+    /**
+        @notice Returns whether deposit id is eligible for upkeep
+        @return bool
+     */
+    function _isUpkeepEligible(uint256 id_) internal view returns (bool) {
+        if (block.timestamp >= recipientInfo[id_].lastUpkeepTimestamp + recipientInfo[id_].paymentInterval) {
+            return true;
+        }
+        return false;
+    }
+
+    /************************
+     * Restricted Setter Functions
+     ************************/
+
+    /**
+        @notice Setter for maxSwapSlippagePercent.
+        @param slippagePercent_ new slippage value as a fraction / 1000.
+    */
+    function setMaxSwapSlippagePercent(uint256 slippagePercent_) external onlyGovernor {
+        if (slippagePercent_ > 1000000) revert YieldStreamer_InvalidAmount();
+        maxSwapSlippagePercent = slippagePercent_;
+    }
+
+    /**
+        @notice Setter for feeToDaoPercent.
+        @param feePercent_ new fee value as a fraction / 1000.
+    */
+    function setFeeToDaoPercent(uint256 feePercent_) external onlyGovernor {
+        if (feePercent_ > 1000000) revert YieldStreamer_InvalidAmount();
```

Comment:
would replace `1000000` with `1e7`. Much easier to parse, since you don't need to count the zeroes.